src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java

```diff
@@ -445,7 +445,10 @@ else if(identifier.equalsIgnoreCase(primaryIdentifierName)){
                     }//end for loop
                     JSONObject amountObj = body.getJSONObject("amount");
                     String currency = amountObj.getString("currency");
-                    Object customDataObj = body.get("customData");
+                    Object customDataObj = null;
+                    if (body.has("customData")) {
```

Comment:
write/ use a util function get or default to do these null checks more efficiently.

Reply:
Can you check now for the utility function?